ompi/mca/coll/base/coll_base_alltoall.c

```diff
@@ -63,56 +63,57 @@ mca_coll_base_alltoall_intra_basic_inplace(const void *rbuf, int rcount,
     ompi_datatype_get_true_extent ( rdtype, &true_lb, &true_ext);
     max_size = true_ext + ext * (rcount-1);
 
+    /* Initiate all send/recv to/from others. */
+    reqs = coll_base_comm_get_reqs(base_module->base_data, 2);
+    if( NULL == reqs ) { err = OMPI_ERR_OUT_OF_RESOURCE; line = __LINE__; goto error_hndl; }
```

Comment:
2 reasons:
- I prefer to have a single return statement for errors
- to get access to the error message.